### PR TITLE
Add support for room pn title override

### PR DESCRIPTION
**Pull request by luismfonseca**
We want to add support for the Push Notifications Title Override.

### CODE REVIEW

File: internal/core/types.go [hunk 1]
```diff
@@ -17,14 +17,15 @@ type User struct {
...
-	Name          string      `json:"name"`                      // Name assigned to the room
-	Private       bool        `json:"private"`                   // Indicates if room is private or not
-	MemberUserIDs []string    `json:"member_user_ids,omitempty"` // List of user id's in the room
-	CustomData    interface{} `json:"custom_data,omitempty"`     // Custom data that can be added to rooms
-	CreatedAt     time.Time   `json:"created_at"`                // Creation timestamp
-	UpdatedAt     time.Time   `json:"updated_at"`                // Updation timestamp
+	ID                            string      `json:"id"`                                         // ID assigned to a room
+	CreatedByID                   string      `json:"created_by_id"`                              // User ID that created the room
+	Name                          string      `json:"name"`                                       // Name assigned to the room
+	PushNotificationTitleOverride *string     `json:"push_notification_title_override,omitempty"` // Optionally overridde Push Notification title
```
Comment (vivangkumar) [inline internal/core/types.go]: override

Comment (luismfonseca) [inline internal/core/types.go]:
```suggestion
	PushNotificationTitleOverride *string     `json:"push_notification_title_override,omitempty"` // Optionally override Push Notification title
```

File: client_test.go [hunk 2]
```diff
@@ -680,16 +691,42 @@ func TestRooms(t *testing.T) {
...
 				So(r.ID, ShouldEqual, room.ID)
 				So(r.Name, ShouldEqual, roomName)
+				So(r.PushNotificationTitleOverride, ShouldResemble, &roomPNTitleOverride)
 				So(r.Private, ShouldEqual, true)
 				So(r.MemberUserIDs, shouldResembleUpToReordering, []string{aliceID, bobID})
 				So(r.CustomData, ShouldResemble, map[string]interface{}{"foo": "bar"})
 			})
 
-			Convey("and update it", func() {
+			Convey("and update it to something else", func() {
```
Comment (jonathanlloyd) [inline client_test.go]: Do these test names make sense? The parent test is talking about room IDs

Comment (luismfonseca) [reply client_test.go]: I think they still do, sort of.

> we can create a room without providing an ID
>> and get it
>> and update it to something else

Maybe the parent should be more:
> we can create a room (without providing an ID)

**Pull request merged**